tribe-ext-event-progress-bar.php

```diff
@@ -0,0 +1,268 @@
+ *     it under the terms of the GNU General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     any later version.
+ *
+ *     This plugin is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *     GNU General Public License for more details.
+ */
+
+namespace Tribe\Extensions\EventProgressBar;
+
+use Tribe__Extension;
+
+/**
+ * Define Constants
+ */
+
+if ( ! defined( __NAMESPACE__ . '\NS' ) ) {
+	define( __NAMESPACE__ . '\NS', __NAMESPACE__ . '\\' );
+}
+
+// Do not load unless Tribe Common is fully loaded and our class does not yet exist.
+if (
+	class_exists( 'Tribe__Extension' )
+	&& ! class_exists( NS . 'Main' )
+) {
+	/**
+	 * Extension main class, class begins loading on init() function.
+	 */
+	class Main extends Tribe__Extension {
+
+		/**
+		 * Is Events Calendar PRO active. If yes, we will add some extra functionality.
+		 *
+		 * @return bool
+		 */
+		public $ecp_active = false;
+
+		/**
+		 * Is Event Tickets active. If yes, we will add some extra functionality.
+		 *
+		 * @return bool
+		 */
+		public $et_active = false;
+
+		/**
+		 * Setup the Extension's properties.
+		 *
+		 * This always executes even if the required plugins are not present.
+		 */
+		public function construct() {
+			$this->add_required_plugin( 'Tribe__Events__Main', '5.0' );
+		}
+
+		/**
+		 * Extension initialization and hooks.
+		 */
+		public function init() {
+			// Load plugin textdomain
+			load_plugin_textdomain( 'tribe-ext-event-progress-bar', false, basename( dirname( __FILE__ ) ) . '/languages/' );
+
+			if ( ! $this->php_version_check() ) {
+				return;
+			}
+
+			if ( ! $this->is_using_compatible_view_version() ) {
+				return;
+			}
+
+			add_action( 'tribe_template_after_include:events/list/event/description', [ $this, 'progressbar' ], 10, 3 );
+			add_action( 'tribe_template_after_include:events/day/event/description', [ $this, 'progressbar' ], 10, 3 );
+			add_action( 'wp_enqueue_scripts', [ $this, 'safely_add_scripts' ] );
+		}
+
+		/**
+		 * Compiles the data for the Progress Bar.
+		 *
+		 * @param $file
+		 * @param $name
+		 * @param $template
+		 */
+		public function progressbar( $file, $name, $template ) {
+
+			$event = tribe_get_event( get_the_ID() );
+
+			/**
+			 * Leaving here for testing purposes.
+			 *
+			 * @todo Remove before publishing
+			 */
+			/*
+			echo date('Z' );
+			echo '<br>';
+			echo strtotime('now');
+			echo '<br>';
+			echo strtotime( $event->end_date_utc );
+			echo '<br>';
+			echo strtotime('now') - strtotime( $event->end_date_utc );
+			echo '<br>';
+			echo $event->end_date_utc;*/
+
+			// Bail if event (start) is not in the past, so event hasn't started yet
+			if ( ! $event->is_past ) {
+				/* ?>
+				<div>Event is coming.</div>
+				<?php */
```

Comment:
Remove commented out code?

Reply:
Use TEC's `tribe_is_past_event()`?

Reply:
Consider how https://github.com/moderntribe/the-events-calendar/blob/5.2.1/src/Tribe/Template/Month.php#L1172-L1173 determines if an event is in the future - might be the same logic you need for "if has not yet started".

Reply:
On `tribe_is_past_event()`:

`tribe_is_past_event()` works differently vs. `! $event->is_past`

`tribe_is_past_event()` is true when the event end date is in the past.
`$event->is_past` is true when the event start date is in the past.

In reference:
`tribe_is_past_event()` is false when the event has started (start date in past) but has not finished yet (end date in future). If we bail then, then the progress bar will not show up for the event that is currently live.

`$event->is_past` is false when the start date AND then end date are in the future.

As a consequence `! $event->is_past` will bail (not show the progress bar) for events whose start date is in the future (not in the past).

`! tribe_is_past_event()` will bail (not show the progress bar) for events that are currently live, where we actually need them to show.